Tests/Factory/PsrHttpFactoryTest.php

```diff
@@ -243,4 +243,20 @@ public function testUploadErrNoFile()
         $this->assertSame(\UPLOAD_ERR_NO_FILE, $uploadedFiles['f1']->getError());
         $this->assertSame(\UPLOAD_ERR_NO_FILE, $uploadedFiles['f2']->getError());
     }
+
+    public function testJsonContent()
+    {
+        $headers = [
+            'HTTP_HOST' => 'http_host.fr',
+            'CONTENT_TYPE' => 'application/json',
+        ];
+        $request = new Request([], [], [], [], [], $headers, '{"city":"Paris","country":"France"}');
+        $psrRequest = $this->factory->createRequest($request);
+
+        if (method_exists(Request::class, 'getPayload')) {
```

Comment:
We should avoid this kind of branching inside a test. I suggest we simply skip this test if the `getPayload()` method is not defined.

Reply:
Done

Reply:
Move the skip condition to the very beginning of the method please.

Reply:
Done

Reply:
Fixed